nexus/src/app/instance.rs

```diff
@@ -731,6 +590,10 @@ impl super::Nexus {
 
                     return Ok(state);
                 }
+                // If there's a previous VMM left behind by an unwinding start
+                // saga, no big deal. We can still start the instance, and the
+                // start saga will remove the dead VMM, allowing it to be reaped.
+                DbVmmState::SagaUnwound => {}
                 DbVmmState::Stopped => {
```

Comment:
I think this routine may need to handle the Stopped case differently now. It seems to be assuming that when sled agent says a VMM has stopped, then Nexus will unlink it from the instance record. That's no longer true, though, because we need an update saga to run first to do that reaping (and we also need the VMM to get to Destroyed first). Seems like this case should return something to the tune of "the instance needs to finish stopping before it can be started." WDYT?

Reply:
Yeah, I think you're right.

Reply:
FWIW, the new logic in `InstanceAndVmm::effective_state` that I've added in this branch (see https://github.com/oxidecomputer/omicron/blob/2ea3fe637bf24667dcddf31a23a7b4813764d426/nexus/db-queries/src/db/datastore/instance.rs#L114-L174) would have done the Correct Thing here, since we recast `Stopped` instances that are still linked to a VMM as `Stopped`, here:
https://github.com/oxidecomputer/omicron/blob/2ea3fe637bf24667dcddf31a23a7b4813764d426/nexus/db-queries/src/db/datastore/instance.rs#L144-L151